ApplicationLibCode/ProjectDataModel/RiaOpmParserTools.cpp

```diff
@@ -106,36 +138,79 @@ std::vector<Opm::VFPProdTable> RiaOpmParserTools::extractVfpProductionTables( co
 
         for ( auto kw : keywordList )
         {
-            auto name = kw->name();
+            auto table = createProductionTable( *kw );
+            tables.push_back( table );
+        }
+    }
+    catch ( ... )
+    {
+    }
 
-            Opm::UnitSystem unitSystem;
-            {
-                const auto& header = kw->getRecord( 0 );
+    return tables;
+}
 
-                if ( header.getItem<Opm::ParserKeywords::VFPPROD::UNITS>().hasValue( 0 ) )
-                {
-                    std::string units_string;
-                    units_string = header.getItem<Opm::ParserKeywords::VFPPROD::UNITS>().get<std::string>( 0 );
-                    unitSystem   = Opm::UnitSystem( units_string );
-                }
-            }
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+std::pair<std::vector<Opm::VFPProdTable>, std::vector<Opm::VFPInjTable>> extractVfpTablesFromDataFile( const std::string& dataDeckFilename )
+{
+    if ( !std::filesystem::exists( dataDeckFilename ) ) return {};
 
-            bool              gaslift_opt_active = false;
-            Opm::VFPProdTable table( *kw, gaslift_opt_active, unitSystem );
-            tables.push_back( table );
+    std::vector<Opm::VFPProdTable> prodTables;
+    std::vector<Opm::VFPInjTable>  injTables;
+
+    try
+    {
+        Opm::Parser parser( false );
+
+        // Required to include the TUNING keyword to avoid parsing error of a Norne DATA file containing the TUNING keyword
+        // The TUNING keyword is not required nor related to VFP data
+        std::vector<Opm::ParserKeyword> parserKeywords = { Opm::ParserKeywords::VFPPROD(),
+                                                           Opm::ParserKeywords::VFPINJ(),
+                                                           Opm::ParserKeywords::INCLUDE(),
+                                                           Opm::ParserKeywords::TUNING() };
+        for ( const auto& kw : parserKeywords )
+        {
+            parser.addParserKeyword( kw );
+        }
+
+        Opm::ParseContext parseContext( Opm::InputError::Action::WARN );
+        auto              deck = parser.parseFile( dataDeckFilename, parseContext );
+
+        {
+            std::string prodKeyword = "VFPPROD";
```

Comment:
Is "VFPPROD" same as Opm::ParserKeywords::VFPPROD?

Reply:
fixed